### PR TITLE
Move defaults from benchmark_spec to vm classes

### PR DESCRIPTION
**Pull request by ehankland**

### CODE REVIEW

File: perfkitbenchmarker/publisher.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def AddMetadata(self, metadata, benchmark_spec):
     metadata = metadata.copy()
     metadata['perfkitbenchmarker_version'] = version.VERSION
     metadata['cloud'] = benchmark_spec.cloud
-    metadata['zones'] = ','.join(benchmark_spec.zones)
+    metadata['zones'] = ','.join(set([vm.zone for vm in benchmark_spec.vms]))
```
Comment (klausw) [inline perfkitbenchmarker/publisher.py]: I was confused by the set() wrapper until I realized it's for making them unique. May deserve a comment.

**Comment by klausw** [comment]
Looks good, but the change description should have a bit more detail since the network changes are nontrivial.

File: perfkitbenchmarker/publisher.py (hunk 1 above)
Comment (ehankland) [inline perfkitbenchmarker/publisher.py]: Done.

**Comment by ehankland** [comment]
Added additional description to the commit message

**Pull request merged**